### DISCUSSION TITLE
`smart`: spacing around dashes

### DISCUSSION THREAD

**Discussion by avidseeker** [jgm/pandoc#10038] [Ideas]
`smart` extension translates `--` and `---` to en and em dashes. E.g:

```
Lorem ipsum ---sed do--- eiusmod -- tempor -- incididunt.
↓
Lorem ipsum —sed do— eiusmod – tempor – incididunt.
```

However, there seems to be different conventions around the spacing of dashes.

* AP Style: Insert a space before and after the em dash, except when used to introduce items in a vertical list.
* Chicago Manual of Style: No spaces around the em dash.
* Newspapers and some magazines: Typically insert a space before and after the em dash.
* Books and journals: Often omit spacing, closing the surrounding text right up next to the em dash.

Can pandoc take an extra layer of abstraction to specify whether to include a space in the output or not?

**Comment by jgm** [comment]
I'd prefer not to complicate things too much by introducing extra options.  It's easy enough to use a Lua filter to add space around all the Em dashes (you'd want nonbreaking spaces in any case, and maybe just partial spaces -- anyway, everyone will want something slightly different).

There's a discussion #10013 about configuring smart quotes.  If we added a way to do that, we could in principle extend it to dashes and ellipses.  But at the moment it's a low priority issue.
